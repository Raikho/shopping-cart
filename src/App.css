.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  /* display: flex; */
  /* flex-direction: column; */
  /* align-items: center; */
  /* justify-content: center; */
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* ============================ CUSTOM ============================= */
/* ================================================================= */
.main-layout {
  width: 100vw;
  height: 100vh;

  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 8fr 1fr;
}

/* ============================ HEADER ============================= */
/* ================================================================= */
.header {
  background-color: rgba(144, 238, 144, 0.192);
  
  display: flex;
  justify-content: space-around;
  align-items: center;
}
.store-title {
  color: dodgerblue;
  font-size: 42px;
  font-weight: bold;
  text-shadow: 4px 4px 4px #000000a8;
}
.nav-container {
  display: flex;
  justify-content: center;
  gap: 20px;
}
a {color: gray;}
a.active {color: lightblue; font-weight: bold;}

/* ============================= MAIN ============================== */
/* ================================================================= */
.shop-layout {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 5fr;
}
.shop-layout>* {
  border: 2px dotted white;
}
.categories {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 20px;
}
.item-list {
  padding: 20px;

  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  justify-items: center;
  align-items: center;
  gap: 20px;
}
.item {
  width: 200px;
  height: 200px;
  box-shadow: 0 0 4px 4px #fff3;

  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 3fr 1fr;
  grid-template-areas: 
    "name name"
    "img img"
    "price buy";
  justify-items: center;
  align-items: center;
  font-size: 20px;

  padding: 10px;
}
.item>.name {grid-area: name;}
.item>img {
  grid-area: img;
  width: 100%;
  height: 100%;
  object-fit: cover;
}
/* ============================ FOOTER ============================= */
/* ================================================================= */
.footer {
  background-color: rgba(230, 212, 173, 0.247);
}




/* 
.App-header {
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 12fr;
}

.App-header>* {
  width: 100%;
  height: 100%;

  display: flex;
  justify-content: center;
  align-items: center;
  gap: 20px;
}

.header {
  background-color: rgba(160, 238, 144, 0.212);
  display: flex;
  flex-direction: center;
  justify-content: space-around;
}
.homepage {
  background-color: rgba(63, 193, 202, 0.199);
}
.shop {
  background-color: rgba(123, 63, 202, 0.199);

  display: grid;
  grid-template-rows: 1fr 5fr;
}
.item-container {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
}
.item {
  width: 200px;
  height: 200px;
  background-color: #61dafb46;
  border-radius: 5px;
  box-shadow: 4px 4px 4px 2px #0003;

  display: flex;
  justify-content: center;
  align-items: center;
}

.store-title {
  color: dodgerblue;
  font-size: 42px;
  font-weight: bold;
  text-shadow: 4px 4px 4px #03171da8;
}
.nav-container {
  display: flex;
  gap: 20px;
}
.cart {
  padding: 5px 10px;
  border-radius: 5px;
  font-weight: bold;
  background-color: dodgerblue;
  color: whitesmoke;
} */