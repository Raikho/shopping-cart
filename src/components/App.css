body {margin: 0; box-sizing: border-box;}
#root {
  width: 100vw;
  height: 100vh;
  margin: 0;
  padding: 0;
  box-sizing: border-box;

  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
  'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: white;
  background-color: #282c34;
  text-align: center;
  font-size: calc(10px + 2vmin);
}

/* ================================================================= */
/* ============================ CUSTOM ============================= */
/* ================================================================= */
:root {
  --button-accent: #0891b2;
  --button-text: white;
  --header-bg: #0008;
  --caption-bg: #0007;

  --visit-fg: black;
  --visit-bg: white;

  --link-inactive: gray;
  --link-active: lightblue;
  --footer-bg: #0008;
}
.App {
  height: 100%;
  width: 100%;

  /* border: 1px dotted yellow; */
}

/* DEBUG */
.content {
  border: 1px dotted lightskyblue;
  padding: 10px;
  overflow-y: auto;

  display: grid;
  grid-template-columns: 1fr;
  justify-items: center;
  gap: 20px;
}
.card {
  background-color: lightcoral;
  height: 200px;
  width: 200px;
}

.footer {
  border: 1px dotted lightgreen;
}
/* =========================== OBJECTS ============================= */
/* ================================================================= */
::-webkit-scrollbar {width: 15px;}
::-webkit-scrollbar-track {background-color: transparent;}
::-webkit-scrollbar-thumb {
  background-color: #d6dee1;
  border-radius: 20px;
  border: 6px solid transparent;
  background-clip: content-box;
}
::-webkit-scrollbar-thumb:hover {background-color: #a8bbbf;}

/* ============================ HEADER ============================= */
/* ================================================================= */
.header {
  border: 1px dotted orange;
  background-color: var(--header-bg);

  display: flex;
  justify-content: space-around;
  align-items: center;
}
.store-title {
  color: dodgerblue;
  font-size: 42px;
  font-weight: bold;
  text-shadow: 4px 4px 4px #000000a8;
}
.nav-container {
  display: flex;
  justify-content: center;
  gap: 20px;
}
a {
  color: var(--link-inactive);
  text-decoration: none;
  text-shadow: 4px 4px 4px #0008;
}
a:hover:not(.active) {filter: brightness(1.3);}
a.active {color: var(--link-active); font-weight: bold;}

/* ========================= MAIN LAYOUTS ========================== */
/* ================================================================= */
.main-layout {
  height: 100%;
  width: 100%;
  display: grid;
  grid-template-rows: 10% 85% 5%;

  background-position: center center;
  background-size: cover;
}
.main-layout.home {background-image: url('../assets/aquarium.jpg');}
.main-layout.shop {background-image: url('../assets/underwater.jpg');}

/* ========================== HOMEPAGE ============================ */
/* ================================================================= */
.homepage {display: flex; justify-content: center; align-items: center;}
.caption {
  display: flex; 
  flex-direction: column;
  justify-content: center; 
  align-items: center;
  padding: 20px;
  gap: 20px;
}
.caption-text {
  text-shadow: 4px 4px 4px #000a;
}
a.enter {
  text-shadow: none;
  padding: 10px 20px;
  border-radius: 2px;
  color: var(--visit-fg);
  background-color: var(--visit-bg);
}
a.enter:hover {
  box-shadow: inset 0 0 2px 2px #0003;
}

/* ========================= SHOP LAYOUT =========================== */
/* ================================================================= */
.shop-layout {overflow-y: auto;}
.categories {
  height: 100px;
  font-size: 32px;

  display: flex;
  justify-content: center;
  align-items: center;
  gap: 20px;
}
@media screen and (max-width: 600px) {
  .categories {flex-direction: column;}
}

/* ============================ ITEMS ============================== */
/* ================================================================= */
.item-list {
  padding: 20px;

  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  justify-items: center;
  align-items: center;
  gap: 20px;
}
.item {
  width: 300px;
  height: 300px;

  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 70% 30%;
  justify-items: start;
  align-items: center;
  font-size: 20px;

}
.item>img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}
.item>.desc {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 3px;
}
.item>.desc>.name {font-size: 20px;}
.item>.desc>.price {font-size: 16px;}

/* =========================== BUTTONS ============================= */
.quantity-container {
  padding: 5px;
  display: flex;
  gap: 5px;
  align-items: center;
}
button {
  user-select: none;
  border-radius: 5px;
  font-size: 16px;
  border: none;
  padding: 5px 10px;
  color: var(--button-text);
  background-color: var(--button-accent);
}
button.add {width: 60px; font-weight: bold;}
button.adjust {
  padding: 3px 9px;
  font-size: 20px;
  color: var(--button-text);
  background-color: var(--button-accent);
}
button:hover {filter: brightness(1.1);}
button:active {filter: brightness(0.9);}

/* ============================= CART ============================== */
/* ================================================================= */

/* ============================ FOOTER ============================= */
/* ================================================================= */
.footer {
  background-color: var(--footer-bg);
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 4px;
}