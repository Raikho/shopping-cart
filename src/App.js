import './App.css';
import { BrowserRouter, Routes, Route, NavLink } from 'react-router-dom';
import Homepage from './Homepage.js';
import Shop from './Shop.js';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <BrowserRouter>
          <div className='header'>
          <NavLink to="/">Home</NavLink>
          <NavLink to="shop">Shop</NavLink>
          </div>
          <Routes>
            <Route path='/' element={<Homepage />} />
            <Route path='shop' element={<Shop />} />
          </Routes>
        </BrowserRouter>
      </header>
    </div>
  );
}

export default App;