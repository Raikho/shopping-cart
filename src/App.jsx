import { useState } from 'react'
import './App.css'

function App() {

  const arr = [1,2,3,4,5,6,7,8,9,10];

  return (
    <div className="App">
      <div className="header">Header</div>
      <div className="content">
        {arr.map(item => (
          <div 
            key={item} 
            className="card"
          >{item}</div>
        ))}
      </div>
      <div className="footer">Footer</div>
    </div>
  )
}

export default App
