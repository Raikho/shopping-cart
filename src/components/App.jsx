import { useState } from 'react'
import './App.css'

function App() {

  const arr = [];
  for (let i = 0; i < 14; i++)
    arr.push(i);

  let isTesting = true;
  const [testHeading, setTestHeading] = useState('Test Heading');
  const clickHandler = () => {setTestHeading('Radical Rhinos');};

  return (
    <div className="App">
      
      {(isTesting) ? 
        <div className="testing">
          <h1>{testHeading}</h1>
          <button type="button" onClick={clickHandler}>Click Me</button>
        </div> :
        null
      }

      <div className="header"></div>
      <div className="content">
        {arr.map(item => (
          <div 
            key={item} 
            className="card"
          >{item}</div>
        ))}
      </div>
      <div className="footer">Footer</div>
    </div>
  )
}

export default App
