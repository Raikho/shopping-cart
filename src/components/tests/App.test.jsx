import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import App from '../App';

// vi.mock('../Header.jsx', () => ({default: vi.fn()}))
// vi.mock('../ItemList.jsx', () => ({
//     default: ({ items, onChangeAmount }) => 
//     <>
//         <button onClick={() => onChangeAmount(items[0].id, 1)}>add</button>
//         <button onClick={() => onChangeAmount(items[0].id, -1)}>subtract</button>
//     </>
// }));

describe.skip('App component', () => {
    it.skip('renders', () => {
        const { container } = render(<App />);
        expect(container).toMatchSnapshot();
    });
});