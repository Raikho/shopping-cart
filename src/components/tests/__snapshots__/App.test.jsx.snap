// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing App component > renders 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="testing"
    >
      <h1>
        Test Heading
      </h1>
      <button
        type="button"
      >
        Click Me
      </button>
    </div>
    <div
      class="header"
    />
    <div
      class="content"
    >
      <div
        class="card"
      >
        0
      </div>
      <div
        class="card"
      >
        1
      </div>
      <div
        class="card"
      >
        2
      </div>
      <div
        class="card"
      >
        3
      </div>
      <div
        class="card"
      >
        4
      </div>
      <div
        class="card"
      >
        5
      </div>
      <div
        class="card"
      >
        6
      </div>
      <div
        class="card"
      >
        7
      </div>
      <div
        class="card"
      >
        8
      </div>
      <div
        class="card"
      >
        9
      </div>
      <div
        class="card"
      >
        10
      </div>
      <div
        class="card"
      >
        11
      </div>
      <div
        class="card"
      >
        12
      </div>
      <div
        class="card"
      >
        13
      </div>
    </div>
    <div
      class="footer"
    >
      Footer
    </div>
  </div>
</div>
`;
